components/dashboard/src/settings/Teams.tsx

```diff
@@ -0,0 +1,767 @@
+                )
+            })}
+        </div>
+        <div className="flex justify-end mt-6">
+            <button className={"ml-2"} onClick={() => props.onClose()}>Close</button>
+        </div>
+    </Modal>);
+}
+
+interface SlotInputHandler {
+    assign: (slot: TeamSubscriptionSlotResolved, assigneeIdentifier: string) => void;
+    reassign: (slot: TeamSubscriptionSlotResolved, newAssigneeIdentifier: string) => void;
+    deactivate: (slot: TeamSubscriptionSlotResolved) => void;
+    reactivate: (slot: TeamSubscriptionSlotResolved) => void;
+}
+
+function Slot(props: {
+    slot: Slot,
+    inputHandler: SlotInputHandler;
+}) {
+
+    const [slot, setSlot] = useState<Slot>(props.slot);
+    const [editMode, setEditMode] = useState<boolean>(false);
+    const [assigneeIdentifier, setAssigneeIdentifier] = useState<string | undefined>();
+    const [errorMsg, setErrorMsg] = useState<string | undefined>();
+
+    useEffect(() => {
+        setEditMode((prev) => {
+            const newEditMode = (prev && !!props.slot.loading) || !!props.slot.errorMsg;
+            return newEditMode;
+        })
+
+        setSlot(props.slot)
+    }, [props.slot])
+
+    useEffect(() => {
+        setErrorMsg(slot.errorMsg);
+    }, [slot])
+
+    const key = `assignee-${props.slot.id}`;
+
+    const renderIdentifier = (assigneeIdentifier: AssigneeIdentifier | undefined) => {
+        if (!assigneeIdentifier) {
+            return undefined;
+        }
+        return `${assigneeIdentifier.identity.authHost}/${assigneeIdentifier.identity.authName}`;
+    };
+
+    const handleAssignment = () => {
+        if (slot.state === 'assigned') {
+            props.inputHandler.reassign(slot, assigneeIdentifier || '');
+        } else {
+            props.inputHandler.assign(slot, assigneeIdentifier || '');
+        }
+    };
+
+    const handleCancel = () => {
+        setEditMode(false);
+        setErrorMsg(undefined);
+        setAssigneeIdentifier(undefined);
+    };
+
+    const handleEdit = () => {
+        setEditMode(true);
+        setErrorMsg(undefined);
+        setAssigneeIdentifier('');
+    };
+
+    const handleDeactivation = () => {
+        props.inputHandler.deactivate(slot);
+    }
+
+    const handleReactivation = () => {
+        props.inputHandler.reactivate(slot);
+    }
+
+
+    const getActions = (slot: Slot) => {
+        const actions: JSX.Element[] = [];
+
+        if (editMode) {
+            actions.push((<button key="ok" className={"ml-2 disabled:opacity-50"} disabled={slot.loading} onClick={() => handleAssignment()}>OK</button>));
+            actions.push((<button key="cancel" className={"ml-2 disabled:opacity-50"} disabled={slot.loading} onClick={() => handleCancel()}>Cancel</button>));
+        } else {
+            const actionDeactivate = (<button key="deactivate" className={"ml-2 disabled:opacity-50 border-red-900 bg-red-500 hover:bg-red-700"} disabled={slot.loading} onClick={() => handleDeactivation()}>Deactivate</button>);
```

Comment:
**suggestion**: This can become a secondary action using the danger button variant.

```suggestion
            const actionDeactivate = (<button key="deactivate" className={"ml-2 disabled:opacity-50 secondary danger"} disabled={slot.loading} onClick={() => handleDeactivation()}>Deactivate</button>);
```